src/daemon/default_vm_image_vault.cpp

```diff
@@ -438,36 +454,47 @@ mp::VMImage mp::DefaultVMImageVault::fetch_image(const FetchType& fetch_type, co
         }
         DeleteOnException image_file{source_image.image_path};
 
-        url_downloader->download_to(info.image_location, source_image.image_path, info.size, LaunchProgress::IMAGE,
-                                    monitor);
+        auto future = QtConcurrent::run([&]() -> VMImage {
```

Comment:
This is a clearly identifiable work unit, can it be extracted out into a proper function rather than local lambda?

Reply:
Yes, sure, it's possible.

Reply:
Thanks